Source/AddOns/DynamicRaceDetection/ThreadMonitor/Monitoring/Threads/ThreadExecutionMonitorDispatcher.cs

```diff
@@ -204,14 +206,23 @@ private static string TryGetClassName(UIntPtr location, UIntPtr objH)
         private string GetDebugInformation(UIntPtr location, UIntPtr objH)
         {
             string sourceInfo = "";
+            List<string> debugInfo = new List<string>();
             if (Configuration.EnableReadWriteTracing)
             {
+                for (int i = 4; i < 15; i++)
+                {
+                    var frame = new StackFrame(i, true);
+                    var info = GetSourceInformation(frame);
+                    if (Regex.Match(info,@"\bMachine.cs\b").Success) { break; }
+                    debugInfo.Add(info);
+                    if (i > 15) { break; }
```

Comment:
This is a for loop and `i` isn't changed. This break isn't necessary.

Reply:
Sorry.. that was a debug statement which got in. will fix it.